src/v/cluster/rm_stm.h

```diff
@@ -153,7 +153,7 @@ class rm_stm final : public persisted_stm {
         // a heap of the first offsets of the ongoing transactions
         absl::btree_set<model::offset> ongoing_set;
         absl::flat_hash_map<model::producer_identity, prepare_marker> prepared;
-        absl::flat_hash_map<model::producer_identity, tx_range> aborted;
+        std::vector<tx_range> aborted;
```

Comment:
you're right, it's also problematic because this vector is reconstructed on restart so it will eventually cause OOM, I've already have a task to this https://app.clubhouse.io/vectorized/story/2204